server/handlers/events_streamer.go

```diff
@@ -475,3 +475,52 @@ func closeAdapterConnections(localMeshAdaptersLock *sync.Mutex, localMeshAdapter
 
 	return map[string]*meshes.MeshClient{}
 }
+
+// swagger:route POST /api/events EventsAPI idUIEventHandler
+// Handle sending UI-generated events to the notification center through the server.
+// responses:
+// 200:
+
+func (h *Handler) UIEventHandler(w http.ResponseWriter, req *http.Request, prefObj *models.Preference, user *models.User, provider models.Provider) {
+	fmt.Println("POST /api/events hit")
+	userID := uuid.FromStringOrNil(user.ID)
+
+	defer func() {
+		_ = req.Body.Close()
+	}()
+
+	var eventDetails map[string]interface{}
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		h.log.Error(ErrRequestBody(err))
+		http.Error(w, ErrRequestBody(err).Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = json.Unmarshal(body, &eventDetails)
+	if err != nil {
+		h.log.Error(models.ErrUnmarshal(err, "event details"))
+		http.Error(w, models.ErrUnmarshal(err, "event details").Error(), http.StatusInternalServerError)
+		return
+	}
+
+	eventBuilder := events.NewEvent().FromUser(userID).FromSystem(*h.SystemID).WithCategory(eventDetails["category"].(string)).
+		WithAction(eventDetails["action"].(string)).WithSeverity(events.EventSeverity(eventDetails["severity"].(string))).
+		WithDescription(eventDetails["description"].(string)).WithMetadata(eventDetails["metadata"].(map[string]interface{}))
+
+	if actedUpon, ok := eventDetails["acted_upon"].(string); ok {
+		eventBuilder = eventBuilder.ActedUpon(uuid.FromStringOrNil(actedUpon))
+	}
+
+	event := eventBuilder.Build()
```

Comment:
What happens if all the event parameters are empty? Will we still send an empty event to Client?

Reply:
makes sense! if the client side request has an empty event I think I should be responding with a bad request.